sqlmesh/core/engine_adapter/base.py

```diff
@@ -456,15 +456,22 @@ def create_view(
             )
         )
 
-    def create_schema(self, schema_name: str, ignore_if_exists: bool = True) -> None:
+    def create_schema(
+        self, schema_name: str, ignore_if_exists: bool = True, warn_on_error: bool = True
+    ) -> None:
         """Create a schema from a name or qualified table name."""
-        self.execute(
-            exp.Create(
-                this=exp.to_identifier(schema_name.split(".")[0]),
-                kind="SCHEMA",
-                exists=ignore_if_exists,
+        try:
+            self.execute(
+                exp.Create(
+                    this=exp.to_identifier(schema_name.split(".")[0]),
+                    kind="SCHEMA",
+                    exists=ignore_if_exists,
+                )
             )
-        )
+        except Exception as e:
+            if not warn_on_error:
+                raise
```

Comment:
```suggestion
                raise e
```

Reply:
why?

Reply:
See https://stackoverflow.com/questions/18001721/raise-with-no-argument

> When you raise without arguments, the interpreter looks for the last exception raised and handled. It then acts the same as if you used raise with the most recent exception type, value and traceback.

Reply:
TIL! Thanks for sharing -- ignore me then.